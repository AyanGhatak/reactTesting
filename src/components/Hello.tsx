import * as React from 'react';
interface Hello {
  props: {
    name: string;
    enthusiasmLevel?: number;
    onIncrement?: () => void;
    onDecrement?: () => void;
  };
}

class Hello extends React.Component implements Hello {
  render() {
    const { name, enthusiasmLevel = 1, onIncrement, onDecrement  } = this.props;
    return (
      <div data-sumo-test="hello">
        <div className="greeeting">
          Hello {name + Array(enthusiasmLevel + 1).join('!')}
        </div>
        <button className="increment" onClick={onIncrement}> + </button>
        <button onClick={onDecrement}> - </button>
      </div>
    );
  }
}
export default Hello;
