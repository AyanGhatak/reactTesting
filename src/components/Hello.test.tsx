import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Hello from './Hello';

// Testing basic component rendering
it('renders the correct text when no enthusiasm level is given', () => {
  const hello = shallow(<Hello name="Ayan" />);
  expect(hello.find('.greeeting').text()).toEqual('Hello Ayan!');
});

it('renders the correct text with an explicit enthusiasm level of 5', () => {
  const hello = shallow(<Hello name="Ayan" enthusiasmLevel={5} />);
  expect(hello.find('.greeeting').text()).toEqual('Hello Ayan!!!!!');
  expect(hello).toMatchSnapshot();
});

// Testing props
xit('Check the props', () => {
  const hello = shallow(<Hello name="Ayan" />);
  expect(hello.prop('name')).toEqual('Ayan');
});

// Full rendering.
it('simulates click events', () => {
  const onIncrement = jest.fn();
  const wrapper = mount((
    <Hello
      name="Ayan"
      onIncrement={onIncrement}
    />
  ));
  wrapper.find('.increment').simulate('click');
  expect(onIncrement.mock.calls.length).toBe(1);
});

// Check for data sumo test.
it('renders the correct text when no enthusiasm level is given', () => {
  const hello = shallow(<Hello name="Ayan" />);
  expect(hello.find('[data-sumo-test="hello"]')).toHaveLength(1);
});
