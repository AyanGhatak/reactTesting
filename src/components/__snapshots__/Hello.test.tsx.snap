// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct text with an explicit enthusiasm level of 5 1`] = `
<div
  data-sumo-test="hello"
>
  <div
    className="greeeting"
  >
    Hello 
    Ayan!!!!!
  </div>
  <button
    className="increment"
  >
     + 
  </button>
  <button>
     - 
  </button>
</div>
`;
